torch/csrc/inductor/aot_inductor_tensor.cpp

```diff
@@ -0,0 +1,185 @@
+#include <torch/csrc/inductor/aot_inductor_tensor.h>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#else
+
+#include <ATen/ops/addmm.h>
+#include <ATen/ops/as_strided.h>
+#include <ATen/ops/empty_strided.h>
+#include <ATen/ops/from_blob.h>
+#include <ATen/ops/mm.h>
+#endif
+
+namespace {
+std::unordered_map<AotInductorScalarType, c10::ScalarType>
+    AotInductorScalarType_to_C10ScalarType = {
+        {kAotInductorByte, c10::kByte},
+        {kAotInductorChar, c10::kChar},
+        {kAotInductorShort, c10::kShort},
+        {kAotInductorInt, c10::kInt},
+        {kAotInductorLong, c10::kLong},
+        {kAotInductorHalf, c10::kHalf},
+        {kAotInductorFloat, c10::kFloat},
+        {kAotInductorDouble, c10::kDouble},
+        {kAotInductorComplexHalf, c10::kComplexHalf},
+        {kAotInductorComplexFloat, c10::kComplexFloat},
+        {kAotInductorComplexDouble, c10::kComplexDouble},
+        {kAotInductorBool, c10::kBool},
+        {kAotInductorBFloat16, c10::kBFloat16},
+};
+
+std::unordered_map<c10::ScalarType, AotInductorScalarType>
+    C10ScalarType_to_AotInductorScalarType = {
+        {c10::kByte, kAotInductorByte},
+        {c10::kChar, kAotInductorChar},
+        {c10::kShort, kAotInductorShort},
+        {c10::kInt, kAotInductorInt},
+        {c10::kLong, kAotInductorLong},
+        {c10::kHalf, kAotInductorHalf},
+        {c10::kFloat, kAotInductorFloat},
+        {c10::kDouble, kAotInductorDouble},
+        {c10::kComplexHalf, kAotInductorComplexHalf},
+        {c10::kComplexFloat, kAotInductorComplexFloat},
+        {c10::kComplexDouble, kAotInductorComplexDouble},
+        {c10::kBool, kAotInductorBool},
+        {c10::kBFloat16, kAotInductorBFloat16},
+};
+
+std::unordered_map<AotInductorDeviceType, c10::DeviceType>
+    AotInductorDeviceType_to_C10DeviceType = {
+        {kAotInductorCPU, c10::kCPU},
+        {kAotInductorCUDA, c10::kCUDA},
+};
+
+std::unordered_map<c10::DeviceType, AotInductorDeviceType>
+    C10DeviceType_to_AotInductorDeviceType = {
+        {c10::kCPU, kAotInductorCPU},
+        {c10::kCUDA, kAotInductorCUDA},
+};
```

Comment:
switch statements might be faster than `std::unordered_map`

Reply:
An even better option might be to use the exact same IDs as eager pytorch, then have some `static_assert`s in this file to make sure they stay in sync.  That way we can convert for free.